src/Auth/Directives.js

```diff
@@ -108,11 +108,34 @@ class AuthenticatedDirective extends SchemaDirectiveVisitor {
   }
 }
 
+class OwnerDirective extends SchemaDirectiveVisitor {
 
+  visitObject(type){
+    this.wrapOwnerAuth(type);
+    type._requiresOwAuth = true;
+  }
+    
+  visitFieldDefinition(field) {
+    const { resolve = defaultFieldResolver } = field;
+    field.resolve = async function (record, args, context, info){
+
+      const OwnerRequester = await getOwnerid(context.token.owner);
+      const OwnerRequested = await getOwnerid(record);
+
+      if(OwnerRequester !== null && OwnerRequested !== null){
```

Comment:
This function will only work on the Profile object.  When you create your comment above the directive in the schema.graphql please ensure that you state it can only be used on the Profile object fields.  We'll add features in the future where this could also be applied on the Teams and Addresses objects.